addons/test_mail/tests/test_message_post.py

```diff
@@ -522,6 +522,22 @@ def test_message_log_with_view(self):
                 }
             )
 
+    def test_mail_message_notify_for_non_thread(self):
```

Comment:
Apparently there already is a `TestNoThread` test class, with a notification test.
In that test, you can just loop over the `with self.assertPostNotifications([{` to call the assert with the subject and without the subject. As the reason it's not currently detected is because we only test with a subject.

```for subject in [False, 'Test Notify']:
       with self.assertPostNotifications([{...}]):
         _message = self.env['mail.thread'].message_notify(..., subject=subject))
```

I think it's good to test on the whole notification flow anyway here, as globally this method is only called in that flow anyway.